File: eTravelAgencija.Model/RequestObjects/ReservationPreviewRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace eTravelAgencija.Model.RequestObjects
{
    public class ReservationPreviewRequest
    {
        public int UserId { get; set; }
        public int OfferId { get; set; }
        public int HotelId { get; set; }
        public int RoomId { get; set; }
        public decimal? BasePrice { get; set; }
        public bool? IncludeInsurance { get; set; }
        public string? VoucherCode { get; set; }
         
    }
}

File: UI/etravel_mobile/lib/models/reservation_preview_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation_preview_request.g.dart';

@JsonSerializable()
class ReservationPreviewRequest {
  final int userId;
  final int offerId;
  final int hotelId;
  final int roomId;
  final double basePrice;
  final bool includeInsurance;
  final String? voucherCode;

  ReservationPreviewRequest({
    required this.userId,
    required this.offerId,
    required this.hotelId,
    required this.roomId,
    required this.basePrice,
    required this.includeInsurance,
    this.voucherCode,
  });

  factory ReservationPreviewRequest.fromJson(Map<String, dynamic> json) =>
      _$ReservationPreviewRequestFromJson(json);

  Map<String, dynamic> toJson() =>
      _$ReservationPreviewRequestToJson(this);
}
